zio-http/src/main/scala/zhttp/service/Server.scala

```diff
@@ -148,6 +159,7 @@ object Server {
   private final case class ConsolidateFlush(enabled: Boolean)                         extends Server[Any, Nothing]
   private final case class AcceptContinue(enabled: Boolean)                           extends UServer
   private final case class FlowControl(enabled: Boolean)                              extends UServer
+  private final case class Allocator(allocator: Option[PooledByteBufAllocator])       extends UServer
```

Comment:
We should use the common base type here which is `ByteBufAllocator`

Reply:
No, because the common base is not implementing ByteBufAllocatorMetricProvider, that basically it gives us access to metrics.